components/ResourceCounts.tsx

```diff
@@ -0,0 +1,79 @@
+import { useEffect, useState } from "react";
+import { Badge, Button } from "@mantine/core";
+import { cleanNotifications, showNotification } from "@mantine/notifications";
+
+export interface ResourceCountResponse {
+  [x: string]: number;
+}
+
+/**
+ * Component which retrieves all resources and their counts, calls on helper functions to sort them by count and translate
+ * them into buttons
+ * @returns array of JSX Buttons
+ */
+const ResourceCounts = () => {
+  const [resources, setResources] = useState<ResourceCountResponse>({});
+  useEffect(() => {
+    fetch(`${process.env.NEXT_PUBLIC_DEQM_SERVER}/resourceCount`)
+      .then((data) => {
+        return data.json() as Promise<ResourceCountResponse>;
+      })
+      .then((resourceCountBody) => {
+        setResources(resourceCountBody);
+      })
+      .catch((error) => {
+        console.log(error.message, "...start the server");
+        cleanNotifications();
+        showNotification({
+          message: "Not connected to server",
+          color: "red",
+          autoClose: false,
+        });
+      });
+  }, []);
+
+  /**
+   * Sorts the resource counts key:value pairs object, then returns them as an array of buttons
+   * @returns array of JSX Buttons that are the sorted resources and their counts
+   */
+  const getResourceCountsNodes = () => {
+    return sortResourceArray(resources).map((el) => (
+      <Button
+        color="cyan"
+        radius="md"
+        size="md"
+        variant="subtle"
+        styles={{
+          inner: {
+            padding: "3px",
+            justifyContent: "flex-start",
+          },
+        }}
+        rightIcon={
+          <Badge color="cyan" data-testid={el}>
+            {resources[el]}
+          </Badge>
+        }
+        key={el}
+      >
+        {" "}
+        {el}{" "}
+      </Button>
+    ));
+  };
+
+  return getResourceCountsNodes();
```

Comment:
Part 2 of the scrollable comment. Let's wrap these buttons in a top-level component that is a [Mantine Stack](https://mantine.dev/core/stack/), which essentially just makes sure that all of its children appear vertically stacked on top of each other. You can play around with the spacing to find something that you like.

On this `<Stack...` component, we should also set the overflow to 'scroll' (using `style` prop) such that when the resource counts panel takes up the full screen height, it scrolls just on the left panel rather than making the app huge